### PR TITLE
feat: adding core carousel

### PR DESCRIPTION
**Pull request by tadjohnston**
[Card](https://rentpath.leankit.com/card/594954095)
 
- added functionality for paging
- added navigation
- added photo carousel

### CODE REVIEW

File: packages/react-ui-core/src/Carousel/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+      className,
+      theme,
+      items,
+      navigation,
+      pagination,
+      showNav,
+      ...rest
+    } = this.props
+
+    console.log('nav: ', this.navigation)
```
Comment (nray621) [inline packages/react-ui-core/src/Carousel/Carousel.js]: leftover console.log

**Comment by mneumark** [comment]
There are a couple of linting issues according to jenkins.

File: packages/react-ui-core/src/Carousel/PhotoCarousel.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  photo(path) {
+    const { server, sizes } = this.props
+    return `${server}${path}${sizes}`
+  }
+
+  lazyLoad() {
+    return (
+      <LazyLoad>
```
Comment (mneumark) [inline packages/react-ui-core/src/Carousel/PhotoCarousel.js]: I see you set up lazyLoad as either a bool or an object, but you never seem to use lazyLoad as an object anywhere. Oversight?

Comment (tadjohnston) [reply packages/react-ui-core/src/Carousel/PhotoCarousel.js]: I meant to add it in as props here, need to check if it's an object here first.

File: packages/react-ui-core/src/Carousel/Carousel.js (hunk 1 above)
Comment (tadjohnston) [reply packages/react-ui-core/src/Carousel/Carousel.js]: Doh! good catch!

**Comment by mneumark** [comment]
Looks good. I see tests are failing because there aren't any, but as far as I know that's getting handled in a different ticket by Beth/Anye. How were you planning to resolve that?

**Comment by nray621** [comment]
LGTM 👍

**Pull request merged**